libenv/sysinfo.c

```diff
@@ -1461,12 +1500,29 @@ static void OSClasses(EvalContext *ctx)
 
     // Extract major version number
     char *major = NULL;
-    for (char *sp = vbuff; *sp != '\0'; sp++)
+    char *minor = NULL;
+    char *sp = vbuff;
+    for (sp; *sp != '\0'; sp++)
     {
         if (major == NULL && isdigit(*sp))
         {
             major = sp;
         }
+        if (major == NULL && !isdigit(*sp))
+        {
+            *sp = '\0';
+        }
+        if (major != NULL && !isdigit(*sp))
+        {
+            break;
+        }
+    }
+    for (sp; *sp != '\0'; sp++)
```

Comment:
The first expression is a statement with no effect. Rather omit it
```suggestion
    for (; *sp != '\0'; sp++)
```